### PR TITLE
NONPR-1718 remove unused mongo dependencies

### PR DESCRIPTION
**Pull request by PaulHodgson**
Second pr for NONPR-1718 which removes the unused `mongo` dependencies.

### CODE REVIEW

**Review by craigtd — APPROVED** [review]
All good (haven't run the tests etc this time though)

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,47 @@
+style = defaultWithAlign
```
Comment (PaulHodgson) [inline .scalafmt.conf:1]: when I updated the other plugins we started seeing moans about this file being missing

File: app/uk/gov/hmrc/nrs/retrieval/config/WSHttp.scala [hunk 2]
```diff
@@ -16,22 +16,6 @@
 
 package uk.gov.hmrc.nrs.retrieval.config
 
-/*
```
Comment (PaulHodgson) [inline app/uk/gov/hmrc/nrs/retrieval/config/WSHttp.scala:19]: removed extra set of header comments

File: build.sbt [hunk 3]
```diff
@@ -17,22 +17,27 @@ lazy val scoverageSettings = {
...
 )
 
+val it = "it"
+
 lazy val test = Seq(
-  "uk.gov.hmrc" %% "hmrctest" % "3.10.0-play-26" % "test",
-  "org.scalatest" %% "scalatest" % "3.0.8" % "test",
-  "com.typesafe.play" %% "play-test" % PlayVersion.current % "test",
-  "org.mockito" % "mockito-all" % "2.0.2-beta" % "test"
+  "uk.gov.hmrc"            %% "bootstrap-test-play-28" % "5.3.0"   % Test,
```
Comment (PaulHodgson) [inline build.sbt:26]: updating the `sbt-plugin` necessitated a bunch of additional upgrades, and `mockito` was a bit of a pain.

File: conf/application.conf [hunk 4]
```diff
@@ -44,8 +44,6 @@ play.http.errorHandler = "uk.gov.hmrc.play.bootstrap.http.JsonErrorHandler"
 # ~~~~
 # Additional play modules can be added here
 
-play.modules.enabled += "play.modules.reactivemongo.ReactiveMongoHmrcModule"
```
Comment (PaulHodgson) [inline conf/application.conf:47]: mongo is not used

File: build.sbt [hunk 5]
```diff
@@ -17,22 +17,27 @@ lazy val scoverageSettings = {
 
 lazy val compile = Seq(
   ws,
-  "uk.gov.hmrc" %% "bootstrap-backend-play-27" % "5.3.0",
-  "uk.gov.hmrc" %% "simple-reactivemongo" % "8.0.0-play-27"
+  "uk.gov.hmrc" %% "bootstrap-backend-play-28" % "5.3.0"
```
Comment (PaulHodgson) [inline build.sbt:20]: mongo is not used

**Review by Leeroyc720 — APPROVED** [review]
LGTM👍

File: test/uk/gov/hmrc/nrs/retrieval/connectors/NonrepRetrievalConnectorSpec.scala [hunk 6]
```diff
@@ -56,12 +54,13 @@ class NonrepRetrievalConnectorSpec extends UnitSpec with MockitoSugar {
   private val connector: NonrepRetrievalConnector = injector.getInstance(classOf[NonrepRetrievalConnector])
 
   "search" should {
-     "make a call to /submission-metadata" in {
+    "make a call to /submission-metadata" in {
       val httpResponseBody: String = "someResponse"
       when(mockHttpResponse.body).thenReturn(httpResponseBody)
-      when(mockHttpResponse.status).thenReturn(200)
+      when(mockHttpResponse.status).thenReturn(OK)
```
Comment (PaulHodgson) [inline test/uk/gov/hmrc/nrs/retrieval/connectors/NonrepRetrievalConnectorSpec.scala:60]: we had moaning about magic numbers

**Review by greg-gajda — APPROVED** [review]
LGTM

**Pull request merged**